ebpf_extensions/neteventebpfext/netevent_ebpf_ext_event.c

```diff
@@ -495,13 +521,13 @@ _ebpf_netevent_push_event(_In_ netevent_event_md_t* netevent_event)
     ebpf_extension_hook_client_t* client_context = NULL;
     netevent_event_notify_context_t netevent_event_notify_context = {0};
     uint64_t event_size = netevent_event->event_data_end - netevent_event->event_data_start;
-    bool push_lock_acquired = false;
+    bool spin_lock_acquired = false;
 
     // Currently, the verifier does not support read-only contexts, so we need to copy the event data, rather than
     // directly passing the existing pointers.
     // Verifier feature proposal: https://github.com/vbpf/ebpf-verifier/issues/639
-    ExAcquirePushLockExclusive(&_ebpf_netevent_push_event_lock);
-    push_lock_acquired = true;
+    KIRQL oldIrql = ExAcquireSpinLockExclusive(&_ebpf_netevent_push_event_lock);
```

Comment:
It looks like ebpf_extension_hook_get_next_attached_client uses a push lock, which cannot be safely used at DISPATCH_LEVEL. The _ebpf_netevent_push_event callback will execute at IRLQ <= DISPATCH_LEVEL, so the design of the callback needs to take this into account. It seems like this was a missed detail when this callback code was originally written